### PR TITLE
make improvements to the typeclass docs based on feedback

### PR DESCRIPTION
**Pull request by stew**

### CODE REVIEW

**Comment by fthomas** [comment]
:+1:

File: docs/src/main/tut/typeclasses.md [hunk 1]
```diff
@@ -37,6 +36,14 @@ implicit val stringShow = new Show[String] {
 log("a string")
 ```
 
+This example demonstrates a powerful property of the typeclass
+pattern, which distinguishes it from a Java interface, which is that
```
Comment (ceedubs) [inline docs/src/main/tut/typeclasses.md]: What do you think about changing this to something like "which distinguishes it from a Java interface that you would make your custom class extend".

My wording isn't great, but here's what's going through my head: often when I'm explaining type classes to someone coming from a Java background, I describe them as being like `Comparator` as opposed to `Comparable`. For people familiar with those two interfaces, this analogy often seems to help them grasp the concept quickly. You can do type classes in Java; it's just very verbose.

Comment (refried) [inline docs/src/main/tut/typeclasses.md]: What about something like:
> This example demonstrates a powerful property of the typeclass pattern.  We have been able to provide an implementation of `Show` for `String`, even though we didn't have the ability change the definition of String (to `extend` a new Java-style interface).  We can use this pattern to retrofit existing types with new behaviors. This is usually referred to as "ad-hoc polymorphism".

Comment (refried) [inline docs/src/main/tut/typeclasses.md]: Or:
> This example demonstrates a powerful property of the typeclass pattern.  We have been able to provide an implementation of `Show` for `String`, without needing to change the definition of `java.lang.String` to `extend` a new Java-style interface; something we couldn't have done even if we wanted to.  We can use the same pattern to retrofit existing types with new behaviors. This is usually referred to as "ad-hoc polymorphism".

**Comment by non** [comment]
:+1:

**Pull request merged**